src/modules/flake-parts/interface.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  flake-parts-lib,
+  ...
+}: let
+  l = lib // builtins;
+  t = l.types;
+in {
+  options = {
+    dream2nix = {
+      lib = l.mkOption {
+        type = t.raw;
+        readOnly = true;
+        description = ''
+          The system-less dream2nix library.
+          This should be the the `lib` attribute of the dream2nix flake.
+        '';
+      };
+      config = l.mkOption {
+        type = t.submoduleWith {
+          modules = [../config];
+        };
+        default = {};
+        description = ''
+          The dream2nix config.
+        '';
+      };
+      projects = l.mkOption {
```

Comment:
I wonder if this is the right time to properly specify what a project is. Especially the `settings` field and how it's merged currently is really hacky and confusing.
Though this might be a bigger task, because some existing code might need to be refactored in order to make it fit to a nicer interface.
Then there is the question if we want to have backward compatibility (and the extra complexity coming with it), or just rip off the band-aid.

Maybe we could merge this, but before we but it into the readme as the `default` way to use dream2nix, we improve/refactor the interface in an extra PR.

Reply:
We can change the name from `projects` to something else, but I'm not sure what would be a better name. Maybe `sources` would be a better name? It would conflict with `source` but I feel like it might be better anyways. Changing the interface should be very simple, after we introduce modules for the arguments of `makeFlakeOutputs` we can just change the type to `t.submoduleWith { modules = [./src/modules/args]; }`. It's just being passed to `makeFlakeOutputs` anyways. So not much is going to change, maybe if we decide on a better name now we can avoid breaking that later though.

Reply:
How about renaming it to `inputs`.

Currently `makeFlakeOutputs` already has an argument `projects` https://github.com/nix-community/dream2nix/blob/2b7456e3d2f0053bc2474fb0c461dd468545277f/examples/debian-binary/flake.nix#L13

So maybe that could lead to confusion.

I think my final goal would be to make a module that is not based on makeFlakeOutputs. It could be designed, to have similar convenience as makeFlakeOuputs, but at the same time allows lower level access, for example specifying the projects manually instead of using the discoverers. 

Some people were mentioning that they have not been very happy with what the discovery phase delivered for them and would prefer to specify the `project` manually. Especially since sometimes, people want to use dream2nix on a single package of their monorepo. And dream2nix just processing everything it sees is not really compatible to that.

Anyways. for now Maybe we just keep it like that, maybe just rename it, and improve it later.

Reply:
`inputs` sounds alright. Also I definitely agree with defining a project manually, but I don't think it really matters here. If we introduce that, nothing really would change in regards to this interface, since we pass everything to the correct dream2nix interface internally. We would continue to define things in the same way and just pass the args to dream2nix, only then we would change the type so that it validates the arguments.

Reply:
`inputs` nor `sources` don't make sense to me compared to `projects` in my opinion. Inputs and sources are things very computing and data specific than `projects` which is more vague. It makes sense that a *project* has *inputs* and *sources*. But I don't see a how `input` == `project.

Reply:
I agree with that, but it's hard to choose a good name because we also use `projects` in the interface currently other than here... I'd rather keep `project` but get rid of the `project` in interface later tbh.

Reply:
Is the plan to move all the interface to flake-parts later on?

Reply:
That would be nice IMO. Since we are already moving internals to modules and we will also move interfaces to modules, there isn't really any reason not to use `flake-parts`. If someone wants to avoid that, it would also be easy since they could just evaluate the modules themselves.

Reply:
Agreed, going full modules is where I'd like to go. We can still have wrappers if required or alternative entry points.